### PR TITLE
doc: document `next('router')` behaviour

### PR DESCRIPTION
**Pull request by HarshithaKP**
Refs: https://github.com/expressjs/expressjs.com/issues/1126

### CODE REVIEW

**Review by dougwilson — CHANGES_REQUESTED** [review]
Thanks! I think we need to get some actual documentation around this feature, though. An example is always good for visual learners, but we should be providing a descriptive section regarding the feature, how it works, etc. and then we can following it with the example you made 👍

**Comment by HarshithaKP** [comment]
@dougwilson, thanks. Described about `next('router')` feature followed by example. PTAL.

**Comment by dougwilson** [comment]
Thank you!

File: _includes/api/en/5x/router-METHOD.md [hunk 1]
```diff
@@ -40,3 +40,27 @@ router.get(/^\/commits\/(\w+)(?:\.\.(\w+))?$/, function (req, res) {
   res.send('commit range ' + from + '..' + to)
 })
 ```
+
+You can use `next` primitive to implement a flow control between different
+middleware functions, based on a specific program state. Invoking `next` with
+the name of a router will cause all the remaining route callbacks on that router
```
Comment (dougwilson) [inline _includes/api/en/5x/router-METHOD.md]:
```suggestion
the string `'router'` will cause all the remaining route callbacks on that router
```

Comment (HarshithaKP) [reply _includes/api/en/5x/router-METHOD.md]: @dougwilson, fixed it. PTAL.

**Review by gireeshpunathil — APPROVED** [review]

**Review by dougwilson — APPROVED** [review]

**Comment by dougwilson** [comment]
squashed & merged

**Pull request merged**